R/make_distance_matrices_list.R

```diff
@@ -0,0 +1,74 @@
+# make_distance_matrices_list.R
+
+# Purpose: This is a helper function for clusterSystems and is unlikely to be useful elsewhere.
+#
+# Parameters:
+# distances           Character vector indicating the distance metrics to be used.
+# all_genes           Character vector of HGNC symbols with which to make distance matrices
+#
+# Value:  A list of objects of class distance.
+
+# Author: Rachel Silverstein
+
+
+make_distance_matrices_list <- function (distances, all_genes) {
+  # initialize a list to hold the distance matrices
+  distanceMatrices <- list()
+
+  for (i in seq_along(distances)) {
+    if (distances[i] == "expression_profile") {
+      GEO <- fetchData("GEOprofiles")
+      expr_dist_matrix <- make_matrix(all_genes, dist_fn =  expr_dist, data_source = GEO)
+      distanceMatrices <- append(distanceMatrices, list(expr_dist_matrix))
+    } else if (distances[i] == "transcription_factor") {
+      GTRD <- fetchData("GTRDgeneTFs")
+      tf_dist_matrix <- make_matrix(all_genes, dist_fn =  tf_dist, data_source = GTRD)
+      distanceMatrices <- append(distanceMatrices, list(tf_dist_matrix))
+    } else if ((distances[i] == "network_jaccard") | (distances[i] == "network_distance")) {
+      STRING <- fetchData("STRINGedges0.8")
+      # convert the STRINGedges object into an igraph object
+      if (requireNamespace("igraph")) {
```

Comment:
I don't know if these are necessary given that they should be installed from the requirements in DESCRIPTION

Reply:
I'm not sure what you mean by "requirements in description" could you explain?

Reply:
`igraph` is mentioned under the `depends:` section in `DESCRIPTION`. Therefore it is redundant to check again whether the package is installed.